### PR TITLE
PMD: start checking UnusedLocalVariable

### PR DESCRIPTION
**Pull request by grimreaper**

### CODE REVIEW

File: code/src/java/pcgen/core/PCClass.java [hunk 1]
```diff
@@ -434,10 +434,9 @@ public void setLevel(final int newLevel, final PlayerCharacter aPC)
 			SubstitutionClassApplication.checkForSubstitutionClass(this, newLevel, aPC);
 		}
 
-		for (PCClass pcClass : aPC.getClassSet())
-		{
-			aPC.calculateKnownSpellsForClassLevel(this);
-		}
+		aPC.getClassSet().stream()
```
Comment (thpr) [inline code/src/java/pcgen/core/PCClass.java]: This feels like it's hiding a bug.  ignored -> this??

Comment (karianna) [reply code/src/java/pcgen/core/PCClass.java]: I think the map can be removed?

Comment (grimreaper) [reply code/src/java/pcgen/core/PCClass.java]: oh, you're right.

**Comment by LegacyKing** [comment]
Is the intent here you remove unused variables, along with enabling the rule?

**Comment by grimreaper** [comment]
> Is the intent here you remove unused variables, along with enabling the rule?

Precisely: enable rule, fix reported issues, and fix any bugs caught as a result.

**Pull request merged**